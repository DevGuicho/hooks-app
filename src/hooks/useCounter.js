import { useState } from "react";

const useCounter = (initialState = 1) => {
  const [counter, setcounter] = useState(initialState);

  const increment = () => {
    setcounter(counter + 1);
  };
  const decrement = () => {
    setcounter(counter - 1);
  };

  const reset = () => {
    setcounter(initialState);
  };
  return { counter, decrement, increment, reset };
};

export default useCounter;
