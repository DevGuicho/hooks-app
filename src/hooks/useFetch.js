import { useEffect, useRef, useState } from "react";

const useFetch = (url) => {
  const isMounted = useRef(true);
  const [state, setstate] = useState({
    data: null,
    isLoading: true,
    error: null,
  });

  useEffect(() => {
    return () => {
      isMounted.current = false;
    };
  }, []);

  useEffect(() => {
    setstate({ data: null, isLoading: true, error: null });
    fetch(url)
      .then((resp) => resp.json())
      .then((data) => {
        if (isMounted.current) {
          setstate({
            isLoading: false,
            error: null,
            data,
          });
        }
      })
      .catch(() => {
        setstate({
          data: null,
          isLoading: false,
          error: "No se pueden cargar la informacion",
        });
      });
  }, [url]);

  return state;
};

export default useFetch;
