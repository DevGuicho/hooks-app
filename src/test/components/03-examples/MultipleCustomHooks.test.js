import { shallow } from "enzyme";
import MultipleCustomHooks from "../../../components/03-examples/MultipleCustomHooks";
import useCounter from "../../../hooks/useCounter";

import useFetch from "../../../hooks/useFetch";
jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("Pruebas en <MultipleCustomHooks />", () => {
  beforeEach(() => {
    useCounter.mockReturnValue({
      counter: 1,
      increment: () => {},
    });
  });

  test("debe de mostrarse correctamente", () => {
    useFetch.mockReturnValue({
      data: null,
      isLoading: true,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar la informaciÃ³n", () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: "Walter White",
          quote: "Say my name",
        },
      ],
      isLoading: false,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find("p").text().trim()).toBe("Say my name");
  });
});
