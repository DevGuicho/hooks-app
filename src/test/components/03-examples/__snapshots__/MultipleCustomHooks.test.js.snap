// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Frase
  </button>
</div>
`;

exports[`Pruebas en <MultipleCustomHooks /> debe mostrar la informaci√≥n 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-right"
  >
    <p>
      Say my name
    </p>
    <footer
      className="blockquote-footer"
    >
      Walter White
    </footer>
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Frase
  </button>
</div>
`;
